R/get_diagnostics.R

```diff
@@ -0,0 +1,125 @@
+#'
+#' Function to retreive model estimates and plot diagnositcs for sdmTMB
+#' index of abundance model runs. 
+#'
+#' @ details
+#' Parameter definitions:
+#' Range: A derived parameter that defines the distance at which two 
+#' points are effectively independent (actually about 13% correlated). 
+#' If the share_range argument is changed to FALSE then the spatial 
+#' and spatiotemporal ranges will be unique, otherwise the default is 
+#' for both to share the same range.
+#' Phi: Observation error scale parameter (e.g., SD in Gaussian).
+#' sigma_O: Standard deviation of the spatial process ("Omega").
+#' sigma_E: Standard deviation of the spatiotemporal process ("Epsilon").
+#'
+#' @template dir  
+#' @param fit List created by the sdmTMB:: fit function.
+#' @param prediction_grid The prediction grid for the survey that sdmTMB will use
+#' to make model predictions to. 
+#'
+#' @import sdmTMB
+#'
+#' @author Chantel R. Wetzel
+#'
+#' @examples
+#' \dontrun{
+#' 	get_diagnostics(
+#' 		sppdir = 'directory location for species specific run', 
+#' 		fit = fit, # object created by sdmTMB::fit function 
+#' 		prediction_grid = year, 
+#' 		survey = survey, # e.g. NWFSC.Combo 
+#' 		obs = 'lognormal')
+#' }
+#'
+#' 
+#'
+#'	
+get_diagnostics <- function(dir, fit, prediction_grid){
```

Comment:
Here or in the fitting function, you may want to run `sanity(fit)`. You can save the output to an object and inspect the elements if you want. `sanity()` just prints stuff. For using in programming: `s <- sanity(fit); all(unlist(s))`. `sanity()` will also return `FALSE` if `fit` is `NULL` or `"try-error"`, which can help with programming.